torch_np/tests/numpy_tests/core/test_multiarray.py

```diff
@@ -5919,9 +5925,10 @@ def test_matrix_matrix_values(self):
         assert_equal(res, tgt12_21)
 
 
-@pytest.mark.xfail(reason='TODO: matmul (ufunc wrapping goes south?)')
 class TestMatmul(MatmulCommon):
-    matmul = np.matmul
+
+    def setup_method(self):
+        self.matmul = np.matmul
```

Comment:
This change is a bit curious. Storing ufuncs as class attributes seems to work ok in scripts/interactive interpreter, but breaks down in combination with pytest:

the argument processing/normalization machinery picks up the extra `self`-like argument, which leads to 

```
>           tensor = torch.as_tensor(obj)
E           RuntimeError: Could not infer dtype of TestMatmul
```

so there's some spooky action on a distance between meta-stuff pytest is doing and what we do here.